libnmstate/validator.py

```diff
@@ -251,9 +251,23 @@ def _assert_vxlan_has_missing_attribute(state, *attributes):
 
 def _assert_vlan_filtering_trunk_tags(ports_state):
     for port_state in ports_state:
-        trunk_tags = port_state.get(LB.Port.VLAN_SUBTREE, {}).get(
-            LB.Port.Vlan.TRUNK_TAGS, []
-        )
+        port_vlan_state = port_state.get(LB.Port.VLAN_SUBTREE, {})
+        port_type = port_vlan_state[LB.Port.Vlan.TYPE]
+        trunk_tags = port_vlan_state.get(LB.Port.Vlan.TRUNK_TAGS, [])
+
+        if port_type == LB.Port.Vlan.ACCESS_TYPE:
+            if trunk_tags:
+                raise NmstateValueError('Access port cannot have trunk tags')
+        else:
+            if len(trunk_tags) == 1:
+                if not trunk_tags[0].get(LB.Port.Vlan.TrunkTags.ID_RANGE):
+                    raise NmstateValueError(
+                        'Trunk port must feature multiple vlan tags'
+                    )
+            elif not trunk_tags:
+                raise NmstateValueError(
+                    'Trunk port must feature multiple vlan tags'
+                )
```

Comment:
Would this be better?
```
try:
    if trunk_tags[0][LB.Port.Vlan.TrunkTags.ID_RANGE]
        continue
except:
    raise NmstateValueError(
        'Trunk port must feature multiple vlan tags'
    )
```

Reply:
I like the idea to avoid the duplicate exception. The problem is that it also needs to be raised if `trunk_tags[0][LB.Port.Vlan.TrunkTags.ID_RANGE]:` is false.

Reply:
Avoiding that duplication would be the icing on the cake, but I'm not sure how to make the code look simpler.

I welcome any other suggestion. Re-open the conversation if you have one.